tests/common.rs

```diff
@@ -107,12 +138,62 @@ macro_rules! tester {
     }};
 }
 
+
+#[cfg(feature = "openssl-tls")]
+#[macro_export]
+macro_rules! openssl_tester {
+    ($f:ident, $endpoint:expr) => {{
+        use std::sync::{Arc, Mutex};
+        use std::thread;
+
+        let c = common::openssl_type($endpoint);
+
+        // we'll need the session_id from the thread
+        // NOTE: even if it panics, so can't just return it
+        let session_id = Arc::new(Mutex::new(None));
+
+        // run test in its own thread to catch panics
+        let sid = session_id.clone();
+        let res = thread::spawn(move || {
+            let mut rt = tokio::runtime::Builder::new()
+                .enable_all()
+                .basic_scheduler()
+                .build()
+                .unwrap();
+            let mut c = rt.block_on(c).expect("failed to construct test client");
+            *sid.lock().unwrap() = rt.block_on(c.session_id()).unwrap();
+            // make sure we close, even if an assertion fails
+            let x = rt.block_on(async move {
+                let r = tokio::spawn($f(c.clone())).await;
+                let _ = c.close().await;
+                r
+            });
+            drop(rt);
+            x.expect("test panicked")
+        })
+        .join();
+        let success = common::handle_test_error(res);
+        assert!(success);
+    }};
+}
+
+#[cfg(feature = "rustls-tls")]
+#[macro_export]
+macro_rules! rustls_local {
+    ($f:ident, $endpoint:expr) => {{
+        let port: u16 = common::setup_server();
+        let f = move |c: Client<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>>| async move { $f(c, port).await };
+        rustls_tester!(f, $endpoint)
+    }};
+}
+
+#[cfg(feature = "openssl-tls")]
 #[macro_export]
-macro_rules! local_tester {
+macro_rules! openssl_local {
     ($f:ident, $endpoint:expr) => {{
         let port: u16 = common::setup_server();
-        let f = move |c: Client| async move { $f(c, port).await };
-        tester!(f, $endpoint)
+        let f = move |c: Client<hyper_tls::HttpsConnector<hyper::client::HttpConnector>>| async move { $f(c, port).await };
```

Comment:
I _think_ this can just be
```rust
let f = move |c| async move { $f(c, port).await };
```

Reply:
Unfortunately the only way I found to do this was to create the closure twice for each connector because it needs a concrete type.